### PR TITLE
Support IPEX bf16 & fp32 optimization for emebedding model 

### PR DESCRIPTION
**Pull request by yuwenzho**
## Type of Change

feature
API changed or not: yes

## Description

Support IPEX bf16 & fp32 optimization for emebedding model by adding a new arg `embedding_precision`.
```python
plugins.retrieval.enable = True
plugins.retrieval.args['embedding_model'] = "hkunlp/instructor-large"
plugins.retrieval.args['precision'] = "bf16"
plugins.retrieval.args["input_path"] = './cnvrg_docs_rag/'
plugins.retrieval.args["persist_dir"] = "./test_dir"
config = PipelineConfig(model_name_or_path="facebook/opt-125m", plugins=plugins)

chatbot = build_chatbot(config)
response = chatbot.predict("Who are the founders of cnvrg.io?")
```

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Review by hshen14 — APPROVED** [review]

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py [hunk 1]
```diff
@@ -71,6 +72,7 @@ def __init__(self,
                  process=True,
                  append=True,
                  polish=False,
+                 embedding_precision=None,
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py:75]: just use precision

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py [hunk 2]
```diff
@@ -93,6 +95,10 @@ def __init__(self,
             "accuracy",
             "general",
         )
+        allowed_embedding_precision: ClassVar[Collection[str]] = (
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py:98]: allowed_precisions

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py (hunk 1 above)
Comment (yuwenzho) [reply intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py]: updated

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py (hunk 2 above)
Comment (yuwenzho) [reply intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py]: updated

**Pull request merged**